### DISCUSSION TITLE
Is ICQ implemented in Hermes?

### DISCUSSION THREAD

**Discussion by 0xphilipp** [informalsystems/hermes#2749] [Q&A] [answered]
Is ICQ already implemented in Hermes or is there a plan to support it?
I do not find much information about the topic of ICQ (ICS-031: https://github.com/cosmos/ibc/tree/main/spec/app/ics-031-crosschain-queries)

**Comment by romac** [comment, marked as answer]
There is ongoing work for ICQ support tracked in https://github.com/informalsystems/hermes/issues/2707 and https://github.com/informalsystems/hermes/issues/2362.

Corresponding PR is at https://github.com/informalsystems/hermes/pull/2709 but seems to be stalled for the moment.

We are currently evaluating how best to proceed with this, and will post updates in the linked issues once we figure out the best course of action.
